bin/jade.js

```diff
@@ -58,6 +59,17 @@ program.on('--help', function(){
 
 program.parse(process.argv);
 
+// locals given, parse them
+
+var locals;
+if (program.locals) {
+  if (exists(program.locals)) {
+    locals = JSON.parse(fs.readFileSync(program.locals));
```

Comment:
I wouldn't bother reviewing the changes in this file too much.  There are lots of changes in master that will need merging with a few changes in this, so what's here at the moment isn't particularly representative of what I expect the finished product to look like.